server/plugin.ts

```diff
@@ -50,20 +57,28 @@ export class AnomalyDetectionOpenSearchDashboardsPlugin
     this.logger = initializerContext.logger.get();
     this.globalConfig$ = initializerContext.config.legacy.globalConfig$;
   }
-  public async setup(core: CoreSetup) {
+  public async setup(
+    core: CoreSetup,
+    { dataSource }: ADPluginSetupDependencies
+  ) {
     // Get any custom/overridden headers
     const globalConfig = await this.globalConfig$.pipe(first()).toPromise();
     const { customHeaders, ...rest } = globalConfig.opensearch;
 
-    // Create OpenSearch client w/ relevant plugins and headers
-    const client: ILegacyClusterClient = core.opensearch.legacy.createClient(
-      'anomaly_detection',
-      {
+    const dataSourceEnabled = !!dataSource;
+
+    let client: ILegacyClusterClient | undefined = undefined;
+
+    if (!dataSourceEnabled) {
+      client = core.opensearch.legacy.createClient('anomaly_detection', {
         plugins: [adPlugin, alertingPlugin],
         customHeaders: { ...customHeaders, ...DEFAULT_HEADERS },
         ...rest,
-      }
-    );
+      });
```

Comment:
Can you add comments about how MDS works, and how the client will be undefined when MDS is available since it is configured/fetched from the datasource plugin based on a data source ID? Making the client optional or updating the type to be `| undefined` can make things easier to track and prevent NPEs down the road.

Reply:
Neo team provided this initialization method, intended to leave client as undefined as this staged. We conditionally create an  `ILegacyClusterClient` only when MDS is not enabled. If MDS is enabled, no client is created because we would like to create the client later based on the request context when we have a dataSourceId. 

Will add more comments in the latest revision

Reply:
Can you add more comments? Don't see any in the latest revision.

Reply:
comment added